[CI] XPackRestIT test {p0=ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations} failing

**Build Scans:**
- [elasticsearch-intake #32879 / rest-compat](https://gradle-enterprise.elastic.co/s/ocdmtgfam5jww)
- [elasticsearch-pull-request #112350 / rest-compatibility](https://gradle-enterprise.elastic.co/s/kbnow345yx5qg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestCompatTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations}" -Dtests.seed=85366A18AB01FB65 -Dtests.locale=szl-Latn-PL -Dtests.timezone=Pacific/Guam -Druntime.java=25
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dml%2F3rd_party_deployment%2FTest%20update%20deployment%20with%20low%20priority%20to%20multiple%20allocations%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [ml/3rd_party_deployment:739]: expected [4xx|5xx] status code but api [ml.update_trained_model_deployment] returned [200 OK] [{"assignment":{"task_parameters":{"model_id":"test_model","deployment_id":"test_model","model_bytes":1630,"threads_per_allocation":1,"number_of_allocations":3,"queue_capacity":10000,"cache_size":"1630b","priority":"low","per_deployment_memory_bytes":0,"per_allocation_memory_bytes":0},"routing_table":{"AYaq9TuaRJeU9fWRMjUS1w":{"current_allocations":1,"target_allocations":3,"routing_state":"started","reason":""}},"assignment_state":"started","start_time":"2026-01-05T20:13:45.297533145Z","max_assigned_allocations":1,"adaptive_allocations":null}}]
```

**Issue Reasons:**
- [9.1] 2 consecutive failures in test test {p0=ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations}
- [9.1] 2 failures in test test {p0=ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations} (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -349,9 +349,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexNodeShutdownIT
   method: testReindexWithShutdown
   issue: https://github.com/elastic/elasticsearch/issues/139806
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations}
-  issue: https://github.com/elastic/elasticsearch/issues/140188
 
 # Examples:
 #

--- a/x-pack/plugin/build.gradle
+++ b/x-pack/plugin/build.gradle
@@ -122,6 +122,7 @@ tasks.named("yamlRestCompatTestTransform").configure({ task ->
   task.skipTest("ml/3rd_party_deployment/Test start and stop deployment with cache", "Deprecated route removed")
   task.skipTest("ml/3rd_party_deployment/Test start and stop multiple deployments", "Deprecated route removed")
   task.skipTest("ml/3rd_party_deployment/Test update model alias on pytorch model to undeployed model", "Deprecated route removed")
+  task.skipTest("ml/3rd_party_deployment/Test update deployment with low priority to multiple allocations", "Low priority deployments must have one allocation")
   task.skipTest("ml/job_cat_apis/Test cat anomaly detector jobs", "Flush API is deprecated")
   task.skipTest("ml/jobs_get_stats/Test get job stats after uploading data prompting the creation of some stats", "Flush API is deprecated")
   task.skipTest("ml/jobs_get_stats/Test get job stats for closed job", "Flush API is deprecated")

